ebos/equil/initstateequil_impl.hh

```diff
@@ -846,18 +859,21 @@ accountForScaledSaturations()
 }
 
 template <class MaterialLawManager, class FluidSystem, class Region, typename CellID>
-double PhaseSaturations<MaterialLawManager, FluidSystem, Region, CellID>::
+std::tuple<double, bool>
+PhaseSaturations<MaterialLawManager, FluidSystem, Region, CellID>::
 applySwatInit(const double pcow)
 {
-    return this->applySwatInit(pcow, this->swatInit_[this->evalPt_.position->cell]);
+    auto [swout, newSwatInit] = this->applySwatInit(pcow, this->swatInit_[this->evalPt_.position->cell]);
+    return {swout, newSwatInit};
 }
 
 template <class MaterialLawManager, class FluidSystem, class Region, typename CellID>
-double PhaseSaturations<MaterialLawManager, FluidSystem, Region, CellID>::
+std::tuple<double, bool>
+PhaseSaturations<MaterialLawManager, FluidSystem, Region, CellID>::
 applySwatInit(const double pcow, const double sw)
 {
-    return this->matLawMgr_
-        .applySwatinit(this->evalPt_.position->cell, pcow, sw);
+    auto [swout, newSwatInit] =this->matLawMgr_.applySwatinit(this->evalPt_.position->cell, pcow, sw);
+    return {swout, newSwatInit};
```

Comment:
Why do you need to change the implementation (body) here?  You're just calling a function with same return type (assuming `Scalar = double`).

Reply:
Same as the above, got some errors and made changes. Reverted back to the original implementation.